database/002_initial_schema.sql

```diff
@@ -1,5 +1,470 @@
+    ON ad
+    FOR UPDATE
+    USING (CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user) and statusCode in ('1', '2'))
+    WITH CHECK ((CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user)) and statusCode in ('1', '2'))
+;
+CREATE POLICY ad_select_policy
+    ON ad
+    FOR SELECT
+    USING (CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user))
+;
+CREATE POLICY ad_ow_policy
+    ON ad
+    FOR ALL
+    TO onlineUsers
+    USING (TRUE)
+    WITH CHECK (TRUE)
+;
+--bookingevent
+CREATE POLICY bookingevent_update_policy
+    ON bookingevent
+    FOR UPDATE
+    USING (statusCode in ('1', '2'))
+    WITH CHECK (statusCode in ('1', '2'))
+;
+CREATE POLICY bookingevent_select_policy
+    ON bookingevent
+    FOR SELECT
+    USING (statusCode in ('1', '2'))
+;
+CREATE POLICY bookingevent_ow_policy
+    ON bookingevent
+    FOR ALL
+    TO onlineUsers
+    USING (TRUE)
+    WITH CHECK (TRUE)
+;
+--itex
+CREATE POLICY itex_update_policy
+    ON itex
+    FOR UPDATE
+    USING (statusCode in ('1', '2'))
+    WITH CHECK (statusCode in ('1', '2'))
+;
+CREATE POLICY itex_select_policy
+    ON itex
+    FOR SELECT
+    USING (statusCode in ('1', '2'))
+;
+CREATE POLICY itex_ow_policy
+    ON itex
+    FOR ALL
+    TO onlineUsers
+    USING (TRUE)
+    WITH CHECK (TRUE)
+;
+--Contracts
+
+CREATE POLICY contracts_update_policy
+    ON contracts
+    FOR UPDATE
+    USING ((CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user)) and statusCode in ('1', '2'))
+    WITH CHECK (statusCode in ('1', '2'))
+;
+CREATE POLICY contracts_select_policy
+    ON contracts
+    FOR SELECT
+    USING (CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user))
+;
+CREATE POLICY contracts_ow_policy
+    ON contracts
+    FOR ALL
+    TO onlineUsers
+    USING(TRUE)
+    WITH CHECK(TRUE)
+;
+-- Queue
+
+CREATE POLICY queue_insert_policy
+    ON queue
+    FOR INSERT
+    WITH CHECK ((CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user)) and statusCode in ('1', '2'))
+;
+CREATE POLICY queue_select_policy
+    ON queue
+    FOR SELECT
+    USING (CID = get_CID_companyUser(current_user) or CID = get_CID_company(current_user))
+;
+CREATE POLICY queue_ow_policy
+    ON queue
+    FOR ALL
+    TO onlineUsers
+    USING (TRUE)
+    WITH CHECK (TRUE)
+;
+
+-- Signatures
+
+CREATE POLICY signatures_ow_policy
+    ON signatures
+    FOR ALL
+    TO onlineUsers
+    USING (TRUE)
+    WITH CHECK (TRUE)
+;
+
+-- Policies for views
+
+--Grant | Saying which columns you can access
+
+--- config
+GRANT CONNECT ON DATABASE booking_dev TO companies,companyUsers,onlineUsers;
+
+
+-- company
+--PS NO Insert on company and companyUsers because they will always added with a role by the admins. -.-' 
+GRANT SELECT (CID,email,username,fullName,orgNum) ON TABLE company TO companyUsers,onlineUsers,companies;
+GRANT DELETE ON TABLE company TO companies;
```

Comment:
Bedkom should be able to remove companies

Reply:
and create them

Reply:
it depends a little on how we decide to add companies to the system. Will they be added only through the companies in OW, only manually or both? (my initial idea is both)

Reply:
Also, as i mentioned above.

I do not think companies should be "users" in themselves

Reply:
I have built the system with the thought that we might want to have companies as users. 

I think we should have both, but i'll need to make som modifications to the database to make that possible i think.

i'll add those permissions to bedkom.